src/lib.rs

```diff
@@ -93,20 +93,53 @@ pub enum Channel {
 ///
 /// [just prints the major and minor versions]: https://github.com/rust-lang/rust/blob/b5c9e2448c9ace53ad5c11585803894651b18b0a/compiler/rustc_codegen_llvm/src/llvm_util.rs#L173-L178
 #[derive(Clone, Debug, Eq, PartialEq, Ord, PartialOrd, Hash)]
-pub struct LLVMVersion {
+pub struct LlvmVersion {
     // fields must be ordered major, minor for comparison to be correct
     /// Major version
     pub major: u64,
     /// Minor version
     pub minor: u64,
 }
 
-impl fmt::Display for LLVMVersion {
+impl fmt::Display for LlvmVersion {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         write!(f, "{}.{}", self.major, self.minor)
     }
 }
 
+impl FromStr for LlvmVersion {
+    type Err = LlvmVersionParseError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let mut parts = s
+            .splitn(3, '.')
+            .map(|part| -> Result<u64, LlvmVersionParseError> {
+                match part.as_bytes() {
+                    // check for at least 2 characters to avoid erroring on "0"
+                    [b'0', _, ..] => Err(LlvmVersionParseError::ComponentMustNotHaveLeadingZeros),
```

Comment:
I think this is a very recent language feature, right? Can we do something less elegant that compiles on older Rust? I know there's no defined MSRV here but I know at least one project that depends on rustc-version that tries to work on something as old as 1.34.

Reply:
Removed the slice matching